docs-src/concepts.md

```diff
@@ -112,42 +114,123 @@ TODO
 
 ### GatewayClass
 
-`GatewayClass` is cluster-scoped resource defined by the infrastructure ops.
-This resource represents a category of Gateways that can be instantiated.
+`GatewayClass` ([source code][gatewayclass-src]) is cluster-scoped resource
+defined by the infrastructure ops. This resource represents a class of Gateways
+that can be instantiated. 
+
+[gatewayclass-src]: https://github.com/kubernetes-sigs/service-apis/blob/master/api/v1alpha1/gatewayclass_types.go
+
+> Note: this serves the same function as the `networking.IngressClass` resource.
 
 ```yaml
-apiVersion: networking.x-k8s.io/v1alpha1
 kind: GatewayClass
 metadata:
-  name: from-internet
-controller:                          # links to a custom resource
-  apiGroup: networking.acme.io       # that implements this class. 
-  kind: cloud-lb
-  name: from-internet
+  name: cluster-gateway
+spec:
+  controller: "acme.io/gateway-controller"
 ```
 
-A sample controller file:
+We expect that one or more `GatewayClasses` will be created by the infrastructure
+provider for the user. It allows decoupling of which mechanism (e.g. controller)
+implements the `Gateways` from the user. For instance, an infrastructure
+provider may create two `GatewayClasses` named `internet`, `private` to reflect
+`Gateways` that define Internet-facing vs private, internal applications.
 
 ```yaml
-apiVersion: networking.acme.io/v1alpha1
-kind: CloudLB
+kind: GatewayClass
+metadata:
+  name: internet
+  ...
+---
+kind: GatewayClass
 metadata:
-  name: from-internet
-networkTier: auto
-ipAllocation: auto
+  name: private
+  ...
 ```
 
-We expect that one or more GatewayClasses will be created by the infrastructure
-provider for the user as part of their cloud infrastructure. If the cluster
-operator wishes to customize their `GatewayClasses`, they are free to do so,
-interacting with the custom resource parameters. Example parameters
-`GatewayClass.controller` resource will include cluster-level defaults and
-constraints.
+The user of the classes will not need to know *how* `internet` and `private` are
+implemented, whether they are cluster-based or cloud-based LBs.
```

Comment:
I think we should stay away from the direct Gateway<>LB association b/c a proxy or other implementation names can be used for a Gateway:
```
The user of the classes will not need to know *how* `internet` and `private` are implemented or whether they are cluster-based or cloud-based.
```

or maybe refer to LBs as an example.

Reply:
done -- made this more generic